andela/travel_tool_front
# #161177585 Fix vacant spaces for guest house

#### What does this PR do?
This PR fixes the number of vacant spaces shown on the guest house details page. It was showing a hard-coded value but it should show the dynamic number of available spaces.

#### Description of Task to be completed?
- Filter rooms by faulty room.
- Filter vacant spaces by booked beds.

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Enter into the project directory - `cd travel_tool_front`
- Install dependencies and start the  app - `make start`
-  [Start](https://github.com/andela/travel_tool_back/pull/139)  the backend.
- Navigate to [travela](http://travela-local.andela.com:3000) in your browser and login
- Go to the `DB` and on the `UserRoles` table, set yourself as a `Travel Admin`
- Go to the browser, refresh the page, Navigate to the `Residence` tab and  click on `Manage`
- You should view the available spaces in the `Available Spaces` card.
- Go the `DB` and set one room bed as booked is true to simulate that someone has booked it.
- The vacant spaces should reduce by one.
- You can also approve a travel request if you are logged in as a manager and the vacant spaces should change accordingly.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#161177585](https://www.pivotaltracker.com/story/show/161177585)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-10-30 at 10 33 03" src="https://user-images.githubusercontent.com/12511966/47702847-396fc500-dc2f-11e8-9b1e-64b0bdfc19b4.png">


#### Questions:
N/A
